BUILD.bazel

```diff
@@ -464,49 +467,25 @@ cc_grpc_library(
 
 cc_library(
     name = "ray_common",
-    srcs = glob(
-        [
-            "src/ray/common/**/*.cc",
-        ],
-        exclude = [
-            "src/ray/common/**/*_test.cc",
-        ],
-    ) + [
-        "src/ray/raylet/scheduling/cluster_resource_data.cc",
-        "src/ray/raylet/scheduling/fixed_point.cc",
-        "src/ray/raylet/scheduling/scheduling_ids.cc",
-    ],
-    hdrs = glob(
-        [
-            "src/ray/common/*.h",
-            "src/ray/common/**/*.h",
-        ],
-    ) + [
-        "src/ray/raylet/scheduling/cluster_resource_data.h",
-        "src/ray/raylet/scheduling/fixed_point.h",
-        "src/ray/raylet/scheduling/scheduling_ids.h",
-    ],
     copts = COPTS,
     strip_include_prefix = "src",
     visibility = ["//visibility:public"],
     deps = [
-        ":node_manager_fbs",
-        ":ray_syncer_cc_grpc",
-        "//src/ray/util:util",
         ":stats_metric",
-        "//src/ray/protobuf:common_cc_proto",
+        "//src/ray/common:asio",
+        "//src/ray/common:constants",
+        "//src/ray/common:event_stats",
+        "//src/ray/common:grpc_util",
+        "//src/ray/common:id",
+        "//src/ray/common:memory_monitor",
+        "//src/ray/common:network",
+        "//src/ray/common:ray_config",
+        "//src/ray/common:ray_syncer",
+        "//src/ray/common:status",
+        "//src/ray/common:task_common",
+        "//src/ray/common:test_util",
```

Comment:
fwiw, you can have a `cc_library` in `src/ray/common` that deps on everything, for use cases like this that is including everything (I think it is everything?).

overall, this change looks really good. :)

Reply:
Yes, move this to src/ray/common should also work.
But my feeling here is that we probably should avoid this because otherwise, people might just always use `//src/ray/common:all`.

This PR is only one step breaking down the ray_common and it should be applied to other places where ray_common is used. We should pick the target for each of the libs correctly to make compiling faster when changes happens.